src/plugins/intel_gna/src/optimizer/gna_pass_manager.cpp

```diff
@@ -2436,6 +2506,85 @@ void TransposeWeightsFromNCHWToNHWCPass::run() {
     }
 }
 
+void FuseFullyConnectedWithEltwisePass::run() {
+    // This legacy pass removes the Eltwise (only if it performs SUM op) from between FC and Any.
+    // The Const layer blob data is added to biases blob data of FC layer.
+    // Finally Const and Reshape (if exists) are also removed
+    //
+    // This operation can be illustrated as follows:
+    //
+    // --
+    //  Original:               Result:         Removed:
+    //
+    //    FC                     FC              (Eltwise)
+    //    |                      |               (Const)
+    //  Reshape                 Any              (Reshape)
+    // (optional)  Const
+    //    |       /
+    //  Eltwise(sum)
+    //    |
+    //  Any (e.g. ReLU)
+    //--
+    //
+    // NOTE: This pass is implemented to prevent unnecessary roundtrip to memory (additional layer).
+    // It can be fully removed if corresponding LPT transformation is already implemented.
+
+    OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "FuseFullyConnectedWithEltwisePass");
+    for (auto& l : *pLayers) {
+        if (!LayerInfo(l).isFullyConnected() || !l->outData.size()) {
+            continue;
+        }
+        auto& fc_layer = l;
+
+        auto rsh_layer = GetNextLayer(fc_layer);
+
+        if (!rsh_layer) {
+            continue;
+        }
+
+        // Find Eltwise - two cases are supported:
+        // 1. There is Reshape between FC and Eltwise
+        // 2. Eltwise is right after FC
+        auto elt_layer = LayerInfo(rsh_layer).isReshape()
+            ? GetNextLayer(rsh_layer)
+            : rsh_layer;
+
+        if (!elt_layer) {
+            continue;
+        }
+
+        // If the layer found is not eltwise, we should skip the pass
+        if (!LayerInfo(elt_layer).isEltwise()
+            || !elt_layer->outData.size()
+            || elt_layer->params["operation"] != "sum") {
+            continue;
+        }
+
+        // Find Const layer attached to Eltwise - needed for getting blob data
+        auto elt_const_layer = CNNNetPrevLayerSkipCertain(elt_layer, 0, [](CNNLayerPtr layer) {
+            return !LayerInfo(layer).isConst();
+        });
+
+        // Find (any) layer after Eltwise
+        auto after_elt_layer = CNNNetGetNextLayerSkipCertain(elt_layer, 0, 0, DoNotSkip).first;
+
+        if (LayerInfo(rsh_layer).isReshape()) {
+            // Connect FC with Eltwise if Reshape exists (Reshape will be removed)
+            if (!IsOneDimLayer(rsh_layer) || !RemoveLayer(fc_layer, rsh_layer, elt_layer)) {
+                // Continue if Reshape layer is not one-dim or RemoveLayer failed
```

Comment:
Or you can remove this comment at all.
```suggestion
                // Continue if Reshape has an input with more than one dimension > 1 or if RemoveLayer failed
```

Reply:
Removed.